### PR TITLE
change alsa callbacks to use signed ints

### LINKED ISSUE #97: Alsa callbacks does not return signed ints
Alsa callbacks should return signed ints, since it reads data in signed format.

Success criteria:
- Alsa callback provides signed ints

### PR DESCRIPTION
**Pull request by MHCooke**
closes #97

### CODE REVIEW

**Review by Jambls — APPROVED** [review]

File: include/callbacks.hh [hunk 1]
```diff
@@ -12,7 +12,7 @@ int main () {
     alsa_callback::acb test ("hw:2,0,0");
 
     auto cb =
-        [](std::vector<uint16_t> left, std::vector<uint16_t> right) {
+        [](std::vector<int16_t> left, std::vector<int16_t> right) {
```
Comment (MarkAHarley) [inline include/callbacks.hh:15]: Do these not need to be const references throughout?

Comment (MHCooke) [reply include/callbacks.hh:15]: Good idea. No point copying them when they don't need to be.

File: include/rms/rms.hh [hunk 2]
```diff
@@ -116,32 +116,32 @@ template <std::size_t num_samples> class RMS {
...
     ~RMS() {
         if (thread_alive) {
             thread.join();
         }
     }
 
     /**
-    * Explicitly delete the copy constructor
-    */ 
+     * Explicitly delete the copy constructor
```
Comment (MarkAHarley) [inline include/rms/rms.hh:128]: Why are all these changes to RMS here, also shouldn't we be avoiding destructors and copy constructors etc entirely now?

Comment (MHCooke) [reply include/rms/rms.hh]: fixed

**Review by MarkAHarley — APPROVED** [review]

**Pull request merged**